commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpClientFactory.java

```diff
@@ -195,11 +195,17 @@ private static void setConfigRepository(final JSch jsch, final File sshDir, fina
         }
     }
 
-    private static void addIndentity(final JSch jsch, final IdentityInfo info) throws FileSystemException {
+    private static void addIdentity(final JSch jsch, final IdentityInfo info) throws FileSystemException {
         try {
-            final String privateKeyFile = info.getPrivateKey() != null ? info.getPrivateKey().getAbsolutePath() : null;
-            final String publicKeyFile = info.getPublicKey() != null ? info.getPublicKey().getAbsolutePath() : null;
-            jsch.addIdentity(privateKeyFile, publicKeyFile, info.getPassPhrase());
+            if (info.getPrivateKey() != null) {

```

Comment:
It is confusing if more than one setting is used but one overrides the other silently. Wouldn't throwing an IllegalStateException be clearer?

Reply:
Sorry, I didn't catch what do you mean by that. Throw` IllegalStateException` in which case?

Reply:
What should happen if the `info` contains BOTH a private key file and private key bytes? Right now the file quietly overrides the bytes. This is good for compatibility. If neither are set you get an exception, which is new behavior. 
Should we throw an exception if both are set?
Is it possible for the provider to be used _without_ a PK?

Reply:
Ah, I see. The issue I have now is that you have to know how which ctor you called to know which getters are valid, which is semi-lame. Let me see if I can adapt your code...